pkg/operator/hostetcdendpointcontroller/members.go

```diff
@@ -41,19 +43,31 @@ func (h *healthyEtcdMemberGetter) GetHealthyEtcdMembers() ([]string, error) {
 	}
 	hostnames := make([]string, 0)
 	for _, m := range member {
-		hostname := getEtcdName(m.PeerURLS)
+		hostname, err := getEtcdName(m.PeerURLS)
+		if err != nil {
+			return nil, err
+		}
 		hostnames = append(hostnames, hostname)
 	}
 	return hostnames, nil
 }
 
-func getEtcdName(peerURLs []string) string {
+func getEtcdName(peerURLs []string) (string, error) {
 	for _, peerURL := range peerURLs {
 		if strings.Contains(peerURL, "etcd-") {
-			return strings.TrimPrefix(strings.Split(peerURLs[0], ".")[0], "https://")
+			return strings.TrimPrefix(strings.Split(peerURLs[0], ".")[0], "https://"), nil
```

Comment:
this existed in old code, my bad in not catching it earlier, but can we do?
```
strings.TrimPrefix(strings.Split(peerURL, ".")[0], "https://"), nil
```

Reply:
s/peerURLs[0]/peerURL/

Reply:
yeah gotcha

Reply:
done